### PR TITLE
Limit the metric alerts to at max 1137 lines

### PR DESCRIPTION
**Pull request by martinkunc**
This downloads original metrics, but limits potentially huge alerts to 1137 lines. This is to handle situation when crashlooping service was generating thousands of new alert instances and the resulting metrics were huge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,3 +76,24 @@ There's a tool named `gen_cert_key.py` that can be used to automatically generat
 ```
 gen_cert_file.py kubeconfig.yaml
 ```
+
```
Comment (iNecas) [inline README.md:79]: seems like unrelated change

File: docs/gathered-data.md [hunk 2]
```diff
@@ -159,37 +159,39 @@ Location in archive: config/configmaps/
 See: docs/insights-archive-sample/config/configmaps
 
 
+## ContainerImages
```
Comment (iNecas) [inline docs/gathered-data.md:162]: same here

Comment (natiiix) [reply docs/gathered-data.md:162]: It's a documentation update from #133, I believe. Because I added `GatherContainerImages` in that PR, but I didn't know I should run some command to update the docs.

Comment (mfojtik) [reply docs/gathered-data.md:162]: separate doc updates and code changes to two commits at least... then i won't care much if they go in single PR.

File: pkg/gather/clusterconfig/clusterconfig.go [hunk 3]
```diff
@@ -47,6 +48,10 @@ const (
 	// imageGatherPodLimit is the maximum number of pods that
 	// will be listed in a single request to reduce memory usage.
 	imageGatherPodLimit = 200
+
+	// metricsLinesLimit is the maximal number of lines read from monitoring Prometheus
```
Comment (mfojtik) [inline pkg/gather/clusterconfig/clusterconfig.go:52]: nit: metricsAlertsLinesLimit

File: pkg/gather/clusterconfig/clusterconfig.go [hunk 4]
```diff
@@ -1151,3 +1172,40 @@ func hasContainerInCrashloop(pod *corev1.Pod) bool {
 	}
 	return false
 }
+
+// LineLimitReader returns a Reader that reads from r
+// but stops with EOF after n lines.
+// The underlying implementation is a *LineLimitedReader.
+func LineLimitReader(r io.Reader, n int64) io.Reader { return &LineLimitedReader{r, n} }
```
Comment (mfojtik) [inline pkg/gather/clusterconfig/clusterconfig.go:1179]: do not export functions that are not used outside this package.

Comment (mfojtik) [reply pkg/gather/clusterconfig/clusterconfig.go:1179]: also name this `NewReaderWithLineLimit()` to follow golang conventions

File: pkg/gather/clusterconfig/clusterconfig.go [hunk 5]
```diff
@@ -1151,3 +1172,40 @@ func hasContainerInCrashloop(pod *corev1.Pod) bool {
...
+
+// A LineLimitedReader reads from R but limits the amount of
+// data returned to just N lines. Each call to Read
+// updates N to reflect the new amount remaining.
+// Read returns EOF when N <= 0 or when the underlying R returns EOF.
+type LineLimitedReader struct {
+	R io.Reader // underlying reader
```
Comment (mfojtik) [inline pkg/gather/clusterconfig/clusterconfig.go:1186]: do not use single letter and **exported** struct vars... name this `reader` and the below `maxLinesLimit`... var names are cheap and good for reading.

**Comment by natiiix** [comment]
@mfojtik @iNecas I believe this PR can be closed, now that #148 is about to be merged.
